scripts/markdown_seo_lint.js

```diff
@@ -0,0 +1,219 @@
+            frontMatter: {},
+        }
+    }
+    // Grab the first file in the list and generate
+    // its full path.
+    const file = paths[0];
+    const fullPath = path.resolve(__dirname, file);
+
+    // Check if the path is a directory
+    const isDirectory = fs.statSync(fullPath).isDirectory();
+
+    // Get the file suffix so we can grab the markdown files.
+    const fileParts = file.split(".");
+    const fileSuffix = fileParts[fileParts.length - 1];
+
+    // Ignore auto generated docs.
+    if (file.indexOf("/content/docs/reference/pkg") > -1) {
+        const remaining = paths.slice(1, paths.length);
+        return searchForMarkdown(remaining, result);
+
+    // If the path is a directory we want to add the contents of the directory
+    // to the list.
+    }else if (isDirectory) {
+        const contents = fs.readdirSync(fullPath).map(function(file) {
+            return fullPath + "/" + file;
+        });
+        paths[0] = contents;
+
+        // Flatten the array.
+        const newPaths = [].concat.apply([], paths);
+        return searchForMarkdown(newPaths, result);
+
+    // Else check if the file suffix is a markdown
+    // and add it the resulting file list.
+    }else {
+        if (fileSuffix === "md") {
+            try {
+                // Read the file contents so we can grab the file header.
+                const content = fs.readFileSync(fullPath, "utf8");
+
+                // Grab the file header.
+                const frontMatter = content.match(FRONT_MATTER_REGEX);
+
+                // Remove the dash blocks around the file header.
+                const fContent = frontMatter[0].split("---").join("");
+
+                // Read the yaml.
+                const obj = yaml.load(fContent);
+
+                // Build the front matter error object and add the file path.
+                result.frontMatter[fullPath] = {
+                    error: null,
+                    title: checkPageTitle(obj.title),
+                    metaDescription: checkPageMetaDescription(obj.meta_desc),
+                };
+                result.files.push(fullPath);
+            } catch(e) {
+                // Include the error message in the front matter error object
+                // so we can display it to the user.
+                result.frontMatter[fullPath] = {
+                    error: e.message,
+                };
+                result.files.push(fullPath);
+            }
+        }
+
+        // If there are remaining paths in the list, keep going.
+        const remaining = paths.slice(1, paths.length);
+        if (remaining.length > 0) {
+            return searchForMarkdown(remaining, result);
+        }
+    }
+
+    return result;
+}
+
+// Get a list of markdown files.
+function getMarkdownFileList(parentPath) {
+    const fullParentPath = path.resolve(__dirname, parentPath)
+    const dirs = fs.readdirSync(fullParentPath).map(function(dir) {
+        return parentPath + "/" + dir;
```

Comment:
There are all sorts of problems (security and logical) when joining paths, so typically you should always call `path.Join(parentDir, dir)` instead of just concatenating things yourself.